src/common/ossl_helpers.c

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+#include <oqs/oqs.h>
+#if defined(OQS_USE_OPENSSL)
+#include <openssl/evp.h>
+#include "ossl_helpers.h"
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+static EVP_MD *sha256_ptr, *sha384_ptr, *sha512_ptr,
+       *sha3_256_ptr, *sha3_384_ptr, *sha3_512_ptr,
+       *shake128_ptr, *shake256_ptr;
+
+static EVP_CIPHER *aes128_ecb_ptr, *aes256_ecb_ptr, *aes256_ctr_ptr;
+#endif
+
+void oqs_fetch_ossl_objects(void) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	sha256_ptr = EVP_MD_fetch(NULL, "SHA256", NULL);
+	sha384_ptr = EVP_MD_fetch(NULL, "SHA384", NULL);
+	sha512_ptr = EVP_MD_fetch(NULL, "SHA512", NULL);
+	sha3_256_ptr = EVP_MD_fetch(NULL, "SHA3-256", NULL);
+	sha3_384_ptr = EVP_MD_fetch(NULL, "SHA3-384", NULL);
+	sha3_512_ptr = EVP_MD_fetch(NULL, "SHA3-512", NULL);
+	shake128_ptr = EVP_MD_fetch(NULL, "SHAKE128", NULL);
+	shake256_ptr = EVP_MD_fetch(NULL, "SHAKE256", NULL);
+
+	aes128_ecb_ptr = EVP_CIPHER_fetch(NULL, "AES-128-ECB", NULL);
+	aes256_ecb_ptr = EVP_CIPHER_fetch(NULL, "AES-256-ECB", NULL);
+	aes256_ctr_ptr = EVP_CIPHER_fetch(NULL, "AES-256-CTR", NULL);
+#endif
+}
+
+const EVP_MD *oqs_sha256(void) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	return sha256_ptr;
```

Comment:
...and then check here whether they are NULL here (meaning OQS_init hasn't been called)? Or may even _then_ "automatically" call `OQS_init`?

Reply:
First, in theory, they still may be NULL even if OQS_init is called - so checking OpenSSL DigestInit result is a good idea.

Second and worse, we will get a threading problem here.

Reply:
> Second and worse, we will get a threading problem here.

ACK. Comment withdrawn.